src/job/jobHandler.ts

```diff
@@ -234,21 +234,94 @@ export abstract class JobHandler {
     }
   }
 
+  private async loggingMessage(resp: CommandExecutorResponse): Promise<void> {
+    await this._logger.save(this.currJob._id, `${'(BUILD)'.padEnd(15)}Finished Build`);
+    await this._logger.save(
+      this.currJob._id,
+      `${'(BUILD)'.padEnd(15)}worker.sh run details:\n\n${resp.output}\n---\n${resp.error}`
+    );
+    if (resp.status != 'success') {
+      const error = new AutoBuilderError(resp.error, 'BuildError');
+      await this.logError(error);
+      throw error;
+    }
+  }
+
+  // call this method when we want benchmarks and uses cwd option to call command outside of a one liner.
+  private async callWithBenchmark(command: string, stage: string): Promise<CommandExecutorResponse> {
+    const start = performance.now();
+    const resp = await this._commandExecutor.execute([command], `repos/${this.currJob.payload.repoName}`);
+    await this._logger.save(
+      this.currJob._id,
+      `${'(COMMAND)'.padEnd(15)} ${command} run details in ${this.currJob.payload.repoName}`
+    );
+    const end = performance.now();
+    const update = {
+      [`${stage}StartTime`]: start,
+      [`${stage}EndTime`]: end,
+    };
+    await this._jobRepository.findOneAndUpdateExecutionTime(this.currJob._id, update);
+    return resp;
+  }
+
+  private async exeBuildModified(): Promise<void> {
+    const stages = {
+      ['get-build-dependencies']: 'nextGenBuildExe',
+      ['next-gen-parse']: 'nextGenParserExe',
+      ['next-gen-html']: 'nextGenHTMLExe',
+      ['oas-page-build']: 'nextGenStageExe',
+    };
+
+    const prerequisiteCommands = this.currJob.buildCommands.slice(0, 3);
```

Comment:
are buildCommands constant? CMIIW, but it seems like they can be different per job and should be found by indexOf

Reply:
The `buildCommadns` seem to be constant in order, i.e. prerequisites come first and keep the order following the [preBuildCommands](https://github.com/mongodb/docs-worker-pool/blob/d14e3a702bdd4436369389716bb6ed3be4be36af/src/job/jobHandler.ts#L399). The `make html` is replaced with the first make command, it is treated as the placeholder if I understood the code correctly.

We could use indexOf if the order isn't consistent and could change in the future.

Reply:
seems like we push commands depending on stage/production? https://github.com/mongodb/docs-worker-pool/blob/d14e3a702bdd4436369389716bb6ed3be4be36af/src/job/productionJobHandler.ts#L65 and https://github.com/mongodb/docs-worker-pool/blob/d14e3a702bdd4436369389716bb6ed3be4be36af/src/job/stagingJobHandler.ts#L55

Reply:
That is correct. Now the make commands are diff based on the environment as you have shown above, and for the make commands we are using indexOf which you can find [here](https://github.com/mongodb/docs-worker-pool/blob/60f301993dbfa3f6a80eeb20be99c09cc16d8b80/src/job/jobHandler.ts#L309-L310). We just don't use them for the prerequisite commands, because that order doesn't change.